packages/test/test-end-to-end-tests/src/test/loaderTest.spec.ts

```diff
@@ -184,13 +184,15 @@ describeNoCompat("Loader.request", (getTestObjectProvider) => {
 	it("can create data object using url with second id, having distinct value from default", async () => {
 		const url = await container.getAbsoluteUrl(dataStore2.handle.absolutePath);
 		assert(url, "dataStore2 url is undefined");
-		const testDataStore = await requestFluidObject<TestSharedDataObject2>(loader, url);
+
+		const container2 = await loader.resolve({ url });
+		container2.connect();
 
 		dataStore1._root.set("color", "purple");
 		dataStore2._root.set("color", "pink");
 
 		await provider.ensureSynchronized();
-
+		const testDataStore = await requestFluidObject<TestSharedDataObject2>(loader, url);
```

Comment:
Why move this? This will make the test flaky since testDataStore may or may not have received the op that sets color.